libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -94,20 +99,36 @@ internal class MapboxRerouteController(
     }
 
     override fun reroute(routesCallback: RerouteController.RoutesCallback) {
-        reroute(
-            NavigationRerouteController.RoutesCallback { routes, _ ->
-                routesCallback.onNewRoutes(routes.toDirectionsRoutes())
-            }
-        )
+        reroute { routes, _ ->
+            routesCallback.onNewRoutes(routes.toDirectionsRoutes())
+        }
     }
 
     override fun reroute(callback: NavigationRerouteController.RoutesCallback) {
         interrupt()
         state = RerouteState.FetchingRoute
-        logD(
-            "Fetching route",
-            LOG_CATEGORY
-        )
+        logD("Fetching route", LOG_CATEGORY)
+
+        ifNonNull(
+            directionsSession.routes,
+            tripSession.getRouteProgress()?.routeAlternativeId,
+        ) { routes, routeAlternativeId ->
+            val relevantAlternative = routes.find { it.routeId == routeAlternativeId }
+            if (relevantAlternative != null) {
+                val newList = mutableListOf(relevantAlternative).apply {
+                    addAll(routes.toMutableList().also { it.removeFirstOrNull() })
+                }.distinct()
```

Comment:
Could you first find the index of the alternative that we picked and remove it from the list by the index instead of using `distinct`? I believe that `distinct` is less performant here (transformation to a `Set` and back) and could lead to other surprising side-effects (like hiding a bug with duplicate routes).

Reply:
Looks lie no sense to look for an index, as I already have alternative item

>I believe that distinct is less performant here 

makes sense 

I'll re-done like
```kotlin
               mutableListOf(relevantAlternative).apply {
                    addAll(routes.toMutableList().apply {
                        removeFirst()
                        remove(relevantAlternative)
                    })
                }
```